Задача: Абстракция, абстракция!

piligrim-curious opened the pull request:
**Main:** Тесты;
**Character**: Абстрактный класс;
**Warrior**: Класс воина;
**Archer**: Класс лучника.

piligrim-curious committed:
Задача: Абстракция, абстракция!
Main: Тесты;
Character: Абстрактный класс;
Warrior: Класс воина;
Archer: Класс лучника.

SnurnicynKonstantin commented on src/main/java/school/faang/abstraction_abstraction/Character.java:
```diff
@@ -0,0 +1,37 @@
+        this.name = name;
+        this.strength = 5;
+        this.agility = 5;
+        this.intellect = 5;
```
У тебя создан конструктор у Character, используй его
`super(name, STRENGTH_VALUE, AGILITY_VALUE, INTELLIGENCE_VALUE);`
Плюс цифры лучше константами оформить:
```
    private static final int STRENGTH_VALUE = 5;
    private static final int AGILITY_VALUE = 5;
    private static final int INTELLIGENCE_VALUE = 5;
```
Ну и их использовать для создания в конструкторе.
Это же замечание относится ко всем другим классам

SnurnicynKonstantin commented on src/main/java/school/faang/abstraction_abstraction/Character.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    public Character(String name, Integer strength, Integer agility, Integer intellect) {
+        if (strength < 0 || agility < 0 || intellect < 0) {
```
Здорово, что проверку добавил

SnurnicynKonstantin commented on src/main/java/school/faang/abstraction_abstraction/Warrior.java:
```diff
@@ -0,0 +1,15 @@
+
+public class Warrior extends Character {
+    public Warrior(String name) {
+        super(name, 10, 5, 3);
```
См. коммент про константы выше

piligrim-curious committed:
Задача: Абстракция, абстракция!
Main: Тесты;
Character: Абстрактный класс;
Warrior: Класс воина;
Archer: Класс лучника.

Изменения:
(Character.java)
* Добавлены константы (используются в конструкторе).
(Warrior.java)
* Добавлены константы.
(Archer.java)
* Добавлены константы.

SnurnicynKonstantin commented on src/main/java/school/faang/abstraction_abstraction/Character.java:
```diff
@@ -0,0 +1,41 @@
+        this.name = name;
+        this.strength = STRENGTH_VALUE;
+        this.agility = AGILITY_VALUE;
+        this.intellect = INTELLIGENCE_VALUE;
```
А здесь лучше использовать конструкор этого же класса:
`this(name, STRENGTH_VALUE, AGILITY_VALUE, INTELLIGENCE_VALUE)`

piligrim-curious committed:
Задача: Абстракция, абстракция!
Main: Тесты;
Character: Абстрактный класс;
Warrior: Класс воина;
Archer: Класс лучника.

Изменения:
(Character.java)
* Использован конструктор этого же класса.

SnurnicynKonstantin reviewed (approved):
Отлично! Теперь и super и this удалось заиспользовать
